### PR TITLE
added better default labels to form components

### PR DESCRIPTION
**Pull request by abidlabs**
This PR adds better default labels to form components. Instead of "null", the label now shows the name of the component. For example, the default label for a slider is "Slider": 

image

Of course, these labels can be overridden by the user to say whatever they want. In the `Interface` class, the labels continue to be based off of the parameter names (for the input components) and are "output" or "output X" for the output components.

As part of this work, I also removed any prop whose value is "null" from the config. This could potentially lead to some problems, so it would be great if you could review this @pngwn. For example, I noticed that the `interactive` prop could take `true`, `false`, or `null`, and so I adapted the logic to not depend on `null`.

Also I included some other minor cleanups that I noticed going through the components.py file. 

Closes: #669 #982

### CODE REVIEW

**Review by pngwn — APPROVED** [review]
This is all working fine, the three `interactive` cases (explicit `true`/`false` and implicit autodetection) are working perfectly and the default values look great!

**Comment by abidlabs** [comment]
Yay thanks @pngwn!

File: ui/packages/app/src/components/Checkbox/Checkbox.svelte [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	export let value: boolean = false;
 	export let default_value: boolean = false;
 	export let style: string = "";
-	export let label: string;
```
Comment (FarukOzderim) [inline ui/packages/app/src/components/Checkbox/Checkbox.svelte:7]: Question: How does the label is realized when the user provides one?

File: test/test_utils.py [hunk 2]
```diff
@@ -164,5 +165,24 @@ def test_format_ner_list_empty(self):
         self.assertEqual(format_ner_list(string, groups), result)
 
 
+class TestDeleteNone(unittest.TestCase):
```
Comment (FarukOzderim) [inline test/test_utils.py:168]: Good practice to write a test for it, great!

File: gradio/components.py [hunk 3]
```diff
@@ -2371,7 +2369,11 @@ def get_template_context(self):
         return {"default_value": self.default_value, **super().get_template_context()}
 
 
+############################
```
Comment (FarukOzderim) [inline gradio/components.py:2372]: Better borders 👍

**Review by FarukOzderim — COMMENTED** [review]
LGTM, Good work!

File: ui/packages/app/src/components/Checkbox/Checkbox.svelte (hunk 1 above)
Comment (FarukOzderim) [reply ui/packages/app/src/components/Checkbox/Checkbox.svelte:7]: @pngwn the default label is assigned like this:
`export let label: string = "Checkbox";`

I was wondering how do we assign the label given by the user

Comment (pngwn) [reply ui/packages/app/src/components/Checkbox/Checkbox.svelte:7]: These `export let` statement essentially marks the interface for the component. They are passed in from above : `<Checkbox label="some_label" />`. Same way arguments are marked on a function and passed in by the caller.

**Pull request merged**